tools/squidclient/squidclient.cc

```diff
@@ -177,6 +178,56 @@ shellUnescape(char *buf)
     *d = '\0';
 }
 
+/// [Proxy-]Authorization header producer
+class Authorization
+{
+public:
+    Authorization(const char *aHeader, const char *aDestination):
+        header(aHeader), destination(aDestination), user(nullptr), password(nullptr) {}
+
+    /// finalizes and writes the right HTTP header to the given stream
+    void commit(std::ostream &os);
+
+    std::string header = nullptr; ///< HTTP header name to send
```

Comment:
std::string are complex objects, not raw-pointers. So they do not need explicit initialization.
```suggestion
    std::string header; ///< HTTP header name to send
```

Reply:
Applied.